pandas/core/resample.py

```diff
@@ -1582,26 +1608,56 @@ def _take_new_index(obj, indexer, new_index, axis=0):
         raise ValueError("'obj' should be either a Series or a DataFrame")
 
 
-def _get_range_edges(first, last, offset, closed='left', base=0):
+def _get_range_edges(first, last, offset, closed='left', base=0,
```

Comment:
woa, you really need 2 additional kwargs here? there is something very strange if that is the case. Pls try to simplify things.

Reply:
fixed, no longer needs kwargs. Can take periods now and will infer the adjustment from the input type.